pkg/signatures/cosign_sig_verifier.go

```diff
@@ -513,7 +513,13 @@ func getVerifiedImageReference(signature oci.Signature, image *storage.Image) ([
 	log.Debugf("Retrieving verified image references from the image names [%v] and signature identity %q",
 		image.GetNames(), signatureIdentity)
 	var verifiedImageReferences []string
-	imageNames := protoutils.SliceUnique(append(image.GetNames(), image.GetName()))
+	// We must ensure here that `append` is not called on the result of `image.GetNames()`.
+	// Otherwise, we create a data race caused by concurrent writes when resizing the
+	// underlying data array of the slice.
+	imageNames := protoutils.SliceUnique(image.GetNames())
```

Comment:
So if we should not use `append` and we do use `append` in line 521 then it means that `protoutils.SliceUnique` copies the array?

Reply:
During code review meeting we adapted a test to reproduce the race, then applied this fix and confirmed that the race disappears. See test at https://github.com/stackrox/stackrox/pull/16671/commits/ae5d6878bbd31351125b9d9d0117019f9e974bf4

Reply:
Cool! Good comments on the tests! Thanks for fixing